res/controllers/Numark-Mixtrack-Pro-FX-scripts.js

```diff
@@ -0,0 +1,782 @@
+                MixtrackProFX.unshift();
+            }
+        },
+        isPress: function(channel, control, value) {
+            if (value === 0x7F) {
+                return true;
+            }
+            return false;
+        }
+    });
+
+    this.loop = new components.Button({
+        outKey: "loop_enabled",
+        midi: [0x94 + channel, 0x40],
+        shift: function() {
+            this.inKey = "loop_in_goto";
+            this.input = components.Button.prototype.input;
+        },
+        unshift: function() {
+            this.input = function(channel, control, value, status, group) {
+                if (value !== 0x7F) {
+                    return;
+                }
+                if (engine.getValue(group, "loop_enabled") === 0) {
+                    script.triggerControl(group, "beatloop_activate");
+                } else {
+                    script.triggerControl(group, "beatlooproll_activate");
+                }
+            };
+        },
+        shiftControl: true,
+        sendShifted: true,
+        shiftOffset: 0x01
+    });
+
+    this.loopHalf = new components.Button({
+        midi: [0x94 + channel, 0x34],
+        shiftControl: true,
+        sendShifted: true,
+        shiftOffset: 0x02,
+        shift: function() {
+            this.inKey = "loop_in";
+            this.outKey = "loop_in";
+            this.disconnect();
+            this.connect();
+        },
+        unshift: function() {
+            this.inKey = "loop_halve";
+            this.outKey = "loop_halve";
+            this.disconnect();
+            this.connect();
+        }
+    });
+
+    this.loopDouble = new components.Button({
+        midi: [0x94 + channel, 0x35],
+        shiftControl: true,
+        sendShifted: true,
+        shiftOffset: 0x02,
+        shift: function() {
+            this.inKey = "loop_out";
+            this.outKey = "loop_out";
+            this.disconnect();
+            this.connect();
+        },
+        unshift: function() {
+            this.inKey = "loop_double";
+            this.outKey = "loop_double";
+            this.disconnect();
+            this.connect();
+        }
+    });
+
+    this.scratchToggle = new components.Button({
+        midi: [0x90 + channel, 0x07],
+        shift: function() {
+            this.inKey = "reverseroll";
+            this.outKey = "reverseroll";
+            this.connect();
+            this.input = components.Button.prototype.input;
+        },
+        unshift: function() {
+            this.disconnect();
+            this.input = function(channel, control, value) {
+                if (value !== 0x7F) {
+                    return;
+                }
+                deck.scratchModeEnabled = !deck.scratchModeEnabled;
+                midi.sendShortMsg(this.midi[0], this.midi[1], deck.scratchModeEnabled ? 0x7F : 0x01);
+            };
+
+            midi.sendShortMsg(this.midi[0], this.midi[1], deck.scratchModeEnabled ? 0x7F : 0x01);
```

Comment:
```suggestion
            midi.sendShortMsg(this.midi[0], this.midi[1], deck.scratchModeEnabled ? this.on : this.off);
```
prefer the appropriate variables instead of just blank values. Make the code easier to understand and change in the future.